pkg/statistics/handle/storage/gc.go

```diff
@@ -182,12 +182,12 @@ func ClearOutdatedHistoryStats(sctx sessionctx.Context) error {
 	}
 	count := rows[0].GetInt64(0)
 	if count > 0 {
-		sql = "delete from mysql.stats_meta_history use index (idx_create_time) where create_time <= NOW() - INTERVAL %? SECOND"
+		sql = "delete from mysql.stats_meta_history use index (idx_create_time) where create_time <= NOW() - INTERVAL %? SECOND limit 5000"
```

Comment:
previously, it will clear all the outdated records. the new code limits the count to 5000.
Is Batch statement a better choice?  https://docs.pingcap.com/tidb/stable/non-transactional-dml

Reply:
Fixed